olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -164,84 +199,124 @@ SignInClientResponse AuthenticationClientImpl::FindSingInResponseInCache(
   return *cached_response;
 }
 
+template <>
+SignInUserResponse AuthenticationClientImpl::FindSignInResponseInCache(
+    int status, const std::string& error_message,
+    const AuthenticationCredentials& credentials) {
+  if (status == static_cast<int>(olp::http::ErrorCode::CANCELLED_ERROR)) {
+    return {{status, error_message}};
+  }
+
+  auto cached_response = user_token_cache_->locked(
+      [&](utils::LruCache<std::string, SignInUserResult>& c)
+          -> boost::optional<SignInUserResult> {
+        auto it = c.Find(credentials.GetKey());
+        return it != c.end() ? boost::make_optional(it->value()) : boost::none;
+      });
+
+  if (!cached_response) {
+    return {{status, error_message}};
+  }
+
+  return *cached_response;
+}
+
+template <>
+void AuthenticationClientImpl::StoreSignInResponseInCache(
+    const AuthenticationCredentials& credentials, SignInResult response) {
+  // Cache the response
+  client_token_cache_->locked(
+      [&](utils::LruCache<std::string, SignInResult>& c) {
+        return c.InsertOrAssign(credentials.GetKey(), response);
+      });
+}
+
+template <>
+void AuthenticationClientImpl::StoreSignInResponseInCache(
+    const AuthenticationCredentials& credentials, SignInUserResult response) {
+  // Cache the response
+  user_token_cache_->locked(
+      [&](utils::LruCache<std::string, SignInUserResult>& c) {
+        return c.InsertOrAssign(credentials.GetKey(), response);
+      });
+}
+
 client::CancellationToken AuthenticationClientImpl::SignInClient(
     AuthenticationCredentials credentials, SignInProperties properties,
     SignInClientCallback callback) {
-  auto sign_in_task =
-      [=](client::CancellationContext context) -> SignInClientResponse {
-    {
-      if (!settings_.network_request_handler) {
-        return {{static_cast<int>(http::ErrorCode::IO_ERROR),
-                 "Cannot sign in while offline"}};
-      }
+  auto task = [=](client::CancellationContext context) -> SignInClientResponse {
+    if (!settings_.network_request_handler) {
+      return OfflineError("Cannot sign in while offline");
+    }
 
-      if (context.IsCancelled()) {
-        return {
-            {static_cast<int>(http::ErrorCode::CANCELLED_ERROR), "Cancelled"}};
+    if (context.IsCancelled()) {
+      return Cancelled();
+    }
+
+    client::OlpClient client = CreateOlpClient(settings_, boost::none, 0);
```

Comment:
The question is, what kind of RetrySettings are you using here? If you are using the same RetrySettings as given to the AuthenticationClient then we do double work. OlpClient should only get RetrySettings with no retry and chopped up timeout based on the backoff strategy.

Reply:
We don't specify any retry setting to the AuthenticationClient, we can do that in future.

Reply:
Even worste, we use default RetrySettings created when you create a OlpClientSettings instance. The default values are 3 retries, 200ms initial backdown and 60s timeout.

Reply:
A parameter was added to disable the retry